### DISCUSSION TITLE
Can't open ACSM file in Calibre or convert it

### DISCUSSION THREAD

**Discussion by Chandy8626** [Leseratte10/acsm-calibre-plugin#70] [Q&A]
I've installed the plug-in and imported the account encryption key. The file is listed in my library, but when I try to open it in Calibre, it launches ADE. When I try to convert it, it says, "Could not convert the book because no supported source format was found." Details say "No supported formats available (Available formats: acsm).

**Comment by Leseratte10** [comment]
Opening an ACSM in Calibre will run the default program configured for ACSM files, which is probably set to ADE on your machine. The plugin doesn't run when converting a file, it runs during import. 

So if you already have an ACSM file in the library and only then install the plugin, that won't work. Copy the ACSM file to another folder, delete it from the Calibre library, then re-add it to the library and it should automatically turn into a book. 

If that doesn't work (and still stays as an ACSM file) you'll need to check the logfile and/or post it here. To do that, right-click the Preferences button and choose "Restart in debug mode", then import the ACSM file again, close Calibre, and post the resulting logfile.

**Reply by Chandy8626** [reply]
Thanks for the reply. I remembered the fact that the plug-in works upon importation of the file and got that far before you responded. Now, I’ve got a new problem: It converted the ACSM to EPUB but the EPUB is still encrypted. 

I deleted the EPUB from the library and reloaded it, but it’s still encrypted. 

I really do appreciate your help. What am I missing?

**Reply by Leseratte10** [reply]
This plugin doesn't do anything regarding EPUB decryption. All it does is generate an EPUB or PDF file just like ADE would. If the distributor specified they'd like to have DRM on the file, the plugin will create an eBook file with DRM. 

To get rid of this, you'll need another plugin called DeDRM.

**Reply by Chandy8626** [reply]
I’ve got that one installed as well.

**Reply by Leseratte10** [reply]
Is it configured correctly? Did it manage to auto-import the encryption key from my plugin?

Try exporting the encryption key in my plugin and import it into DeDRM and try again. If that still doesn't work you'll probably have to post a log on the DeDRM Github for them to check - once you get a DRMed EPUB, my plugin has done its job.

**Reply by Chandy8626** [reply]
Thank you. I have a tendency to get tunnel vision when I'm tired and trying to solve a problem, so, late last night, I didn't realize your plug-in had worked and that my issue was with the DeDRM one.

**Comment by sheepdoggie** [comment]
I am unable to open an acsm file. Here is the debug output. Please advise.

ApplicationPaletteChange event ignored
Using calibre Qt style: True
calibre Debug log
calibre 7.16  embedded-python: True
Linux-5.15.0-130-generic-x86_64-with-glibc2.31 Linux ('64bit', 'ELF')
('Linux', '5.15.0-130-generic', '#140~20.04.1-Ubuntu SMP Wed Dec 18 21:35:34 UTC 2024')
Python 3.11.5
Interface language: None
EXE path: /opt/calibre/bin/calibre-debug
Successfully initialized third party plugins: DeACSM (0, 0, 16) && DeDRM (10, 0, 9) && Annotations (1, 17, 13) && Ebook Translator (2, 3, 5)
calibre 7.16  embedded-python: True
Linux-5.15.0-130-generic-x86_64-with-glibc2.31 Linux ('64bit', 'ELF')
('Linux', '5.15.0-130-generic', '#140~20.04.1-Ubuntu SMP Wed Dec 18 21:35:34 UTC 2024')
Python 3.11.5
Interface language: None
EXE path: /opt/calibre/bin/calibre-debug
Successfully initialized third party plugins: DeACSM (0, 0, 16) && DeDRM (10, 0, 9) && Annotations (1, 17, 13) && Ebook Translator (2, 3, 5)
QPA platform: xcb
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 81.55183946488295 x 81.64285714285714
[0.00] Starting up...
[0.00] Showing splash screen...
[0.10] splash screen shown
[0.10] Initializing db...
[0.12] db initialized
[0.12] Constructing main UI...
DEBUG:    0.0 Annotations plugin: Error loading libiMobileDevice. This hasn't worked for a while, and is blacklisted in calibre v3.
DEBUG:    0.0 Annotations plugin: Error is:  No module named 'calibre.devices.idevice'
DEBUG:    0.0 Annotations::reader_app_support.py - loading translations
DEBUG:    0.0 Annotations::common_utils.py - loading translations
DEBUG:    0.0 Annotations::appearance.py - loading translations
DEBUG:    0.0 Annotations::config.py - loading translations
DEBUG:    0.0 Annotations::annotated_books.py - loading translations
DEBUG:    0.1 Annotations::action.py - loading translations
[0.91] GUI main window shown
[1.02] main UI initialized...
[1.02] Hiding splash screen
[1.12] Starting the smartdevice driver
SMART_DEV (   0.00:  0.001) _attach_to_port try ip address "" on port 9090
SMART_DEV (   0.95:  0.806) _startup_on_demand listening on port 9090
SMART_DEV (   0.98:  0.000) _attach_to_port try ip address "" on port 54982
SMART_DEV (   1.00:  0.000) _startup_on_demand broadcast socket listening on port 54982
[2.14] Finished starting smartdevice
DeACSM v0.0.16: Trying to parse file DeathinaStrangeCount_9781555848989_1268247.acsm
DeACSM v0.0.16: Try to fulfill ...
DeACSM v0.0.16: Hey, that didn't work: 
Looks like there's been an error during Fulfillment: <error xmlns="http://ns.adobe.com/adept" data="E_ADEPT_REQUEST_EXPIRED http://acsepub.contentreserve.com/fulfillment/Fulfill 2025-01-04T03:45:22-05:00%20(1735980322000)%20is%20before%202025-01-04T13:57:08-05:00%20(1736017028318)"/>

SMART_DEV (   2.02:  1.001) run All IP addresses {'lo': [{'addr': '127.0.0.1', 'netmask': '255.0.0.0', 'peer': '127.0.0.1'}], 'eno1': [{'addr': '192.168.0.158', 'netmask': '255.255.255.0', 'broadcast': '192.168.0.255'}]}
[23.24] splash screen hidden
[23.24] Started up in 23.24 seconds with 13 books
Added DeathinaStrangeCount 9781555848989 1268247 to db in: 0.1
A
